keras_nlp/layers/modeling/alibi_bias.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.backend import keras
+from keras_nlp.backend import ops
+
+
+@keras_nlp_export("keras_nlp.layers.AlibiBias")
+class AlibiBias(keras.layers.Layer):
+    """A layer that add the alibi bias to attention scores
+
+    This layer generates a linear, non-learned bias. Defined and formalized in
+    [Train Short, Test Long: Attention with Linear Biases Enables Input Length Extrapolation](https://arxiv.org/abs/2108.12409).
+
+    Takes as input an attention score. This layer will return the attention
+    scores after adding the alibi bias to it. The output will have the same
+    shape as the input.
+
+    Args:
+        alibi_bias_max: int. This value will be used to compute the slope of
+            each head. The heads slopes is a geometric sequence that starts at
+            `2**(-alibi_bias_max/num_heads)` and uses that same value as its
+            ratio. Defaults to 8.
+    Call arguments:
+        attention_scores: The result of multipying the query and the key of the
+            multi head attention of the transformer. The shape must be greater
+            than or equal to 3 with the last 3 dimensions equal to
+            `(num_heads, query_length, key_length)`.
+
+    Examples:
+    ```python
+    # create a simple layer that takes token embeddings as input and generates
+    # the alibi tensor
+    seq_len = 100
+    vocab_size = 1000
+    embedding_dim = 32
+    inputs = keras.Input((seq_len,), dtype="float32")
+    embedding = keras.layers.Embedding(
+        input_dim=vocab_size, output_dim=embedding_dim
+    )(inputs)
+    alibi_bias = keras_nlp.layers.AlibiBias(num_heads=8)(embedding)
+    ```
+
+    References:
+     - [Press et al., 2021](https://arxiv.org/abs/2108.12409)
+    """
+
+    def __init__(
+        self,
+        max_sequence_length,
+        alibi_bias_max=8,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.max_sequence_length = max_sequence_length
+        self.alibi_bias_max = alibi_bias_max
+
+    def build(self, inputs_shape):
```

Comment:
For rotary we just simply recompute the vectors each time--no weights!

Assuming that works correctness wise, I would do the same here for consistency.

(again if we discover this is a practical slowdown, we can update both this and the RotaryEmbedding layer together)